search/includes/classes/commands/class-healthcommand.php

```diff
@@ -47,82 +47,124 @@ public function validate_counts( $args, $assoc_args ) {
 	 * [--version=<int>]
 	 * : Index version to validate - defaults to all
 	 *
+	 * [--network-wide]
+	 * : Validate all sites in a multisite network
+	 *
 	 * ## EXAMPLES
 	 *     wp vip-es health validate-users-count
 	 *
 	 * @subcommand validate-users-count
 	 */
 	public function validate_users_count( $args, $assoc_args ) {
-		WP_CLI::line( sprintf( "Validating users count\n" ) );
+		$this->validate_indexable_count( 'user', $assoc_args );
+	}
 
-		$search = \Automattic\VIP\Search\Search::instance();
+	/**
+	 * ## OPTIONS
+	 *
+	 * [--version=<int>]
+	 * : Index version to validate - defaults to all
+	 *
+	 * [--network-wide]
+	 * : Validate all sites in a multisite network
+	 *
+	 * ## EXAMPLES
+	 *     wp vip-es health validate-posts-count
+	 *
+	 * @subcommand validate-posts-count
+	 */
+	public function validate_posts_count( $args, $assoc_args ) {
+		$this->validate_indexable_count( 'post', $assoc_args );
+	}
 
-		$versions = [];
+	/**
+	 * Generic internal function to validate counts on any indexable,
+	 * supporting multisite installations
+	 *
+	 * @param string $indexable_slug Slug of the indexable to validate
+	 * @param array $assoc_args CLI arguments
+	 */
+	private function validate_indexable_count( $indexable_slug, $assoc_args ) {
+		$search = \Automattic\VIP\Search\Search::instance();
 
 		if ( isset( $assoc_args['version'] ) ) {
-			$versions[] = intval( $assoc_args['version'] );
+			$version = intval( $assoc_args['version'] );
 		} else {
-			// Defaults to all versions
-			$indexable = \ElasticPress\Indexables::factory()->get( 'user' );
+			$version = null;
+		}
 
-			$version_objects = $search->versioning->get_versions( $indexable );
+		if ( isset( $assoc_args['network-wide'] ) && is_multisite() && $search->is_network_mode() ) {
```

Comment:
We don't actually need the `$search->is_network_mode()` check - we want to be able to re-index all sites even if we're not using EP's "network mode", which allows sites to cross-search eachother.
```suggestion
		if ( isset( $assoc_args['network-wide'] ) && is_multisite() ) {
```